src/clj/puppetlabs/trapperkeeper/services/watcher/filesystem_watch_core.clj

```diff
@@ -95,6 +95,7 @@
   (let [watch-key (.take (:watch-service watcher))
         events (.pollEvents watch-key)]
     (.reset watch-key)
+    (watch-new-directories! (map #(clojurize % (.watchable watch-key)) events) watcher)
```

Comment:
This same call `(map #(clojurize % (.watchable watch-key)) events)` happens elsewhere. Would it be worth breaking into a utility function?